### DISCUSSION TITLE
How to declare a property that is set in the MSBuild invocation?

### DISCUSSION THREAD

**Discussion by paulomorgado** [mhutch/MonoDevelop.MSBuildEditor#155] [Q&A]
When a property is set in the MSBuild invocation (either explicitly or by environment variable), it's always marked as not assigned:

![image](https://github.com/mhutch/MonoDevelop.MSBuildEditor/assets/470455/3d6e19aa-2c43-4e1a-8b09-7c6eac9e4442)

Is there a way to inform that it's externally set?

**Comment by mhutch** [comment]
Not currently. This class of issues should be mostly solved when I add support for opening files within the context of having been imported by a project, rather than treating every open file as if it's the toplevel.

There will of course be some properties that are expected to be set via an environment variable or commandline argument, and for these you will need a schema file. However, maybe we can simplify this with a special "environment schema" and a context menu "add to environment schema" or something.
